### PR TITLE
Drop support for IE11

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: style/variables.less [hunk 1]
```diff
@@ -12,13 +10,4 @@
...
   }
-
-  // IE
-  &:-ms-input-placeholder {
-    color: @color;
-  }
-  // Edge
-  &::-ms-input-placeholder {
-    color: @color;
-  }
```
Comment (nstepien) [inline style/variables.less]: Since canary won't be used for a while still, I figure we may as well remove those now.

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -265,15 +264,13 @@ function DataGrid<R, K extends keyof R, SR>({
...
               onSort={props.onSort}
-              scrollLeft={nonStickyScrollLeft}
             />
             {enableFilters && (
               <FilterRow<R, SR>
                 height={headerFiltersHeight}
                 width={columnMetrics.totalColumnWidth + getScrollbarSize()}
                 lastFrozenColumnIndex={columnMetrics.lastFrozenColumnIndex}
                 columns={viewportColumns}
-                scrollLeft={nonStickyScrollLeft}
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:276]: Do you think we can remove `scrollLeft` from `InteractionMasks` also?

File: src/masks/InteractionMasks.tsx [hunk 3]
```diff
@@ -114,7 +114,7 @@ export default function InteractionMasks<R, SR>({
   function getEditorPosition() {
     if (!canvasRef.current) return null;
     const { left, top } = canvasRef.current.getBoundingClientRect();
-    const { scrollTop: docTop, scrollLeft: docLeft } = document.scrollingElement || document.documentElement;
+    const { scrollTop: docTop, scrollLeft: docLeft } = document.scrollingElement!;
```
Comment (amanmahajan7) [inline src/masks/InteractionMasks.tsx:117]: Do we need `!`? Can it be null?

File: style/variables.less [hunk 4]
```diff
@@ -2,8 +2,6 @@
 
 .user-select(@value: none) {
   -webkit-user-select: @value;
-  -moz-user-select: @value;
```
Comment (amanmahajan7) [inline style/variables.less:5]: Do we not need `moz-user-select`?

File: src/masks/InteractionMasks.tsx (hunk 3 above)
Comment (nstepien) [reply src/masks/InteractionMasks.tsx:117]: image

Unlikely at best. 🤔

File: src/DataGrid.tsx (hunk 2 above)
Comment (nstepien) [reply src/DataGrid.tsx:276]: I believe we need it to position the editor, and close the editor when scrolling.
We can look into it further in the future.

File: style/variables.less (hunk 4 above)
Comment (nstepien) [reply style/variables.less:5]: It's been unprefixed since Firefox 69.

File: src/masks/InteractionMasks.tsx (hunk 3 above)
Comment (amanmahajan7) [reply src/masks/InteractionMasks.tsx:117]: Although we are not supporting IE, I think it is okay to add back `document.scrollingElement || document.documentElement;` so rdg is somewhat usable in IE. It is not a big change anyway

File: src/masks/InteractionMasks.tsx [hunk 5]
```diff
@@ -276,12 +276,7 @@ export default function InteractionMasks<R, SR>({
     e.dataTransfer.effectAllowed = 'copy';
     // Setting data is required to make an element draggable in FF
     const transferData = JSON.stringify(selectedPosition);
-    try {
-      e.dataTransfer.setData('text/plain', transferData);
-    } catch (ex) {
-      // IE only supports 'text' and 'URL' for the 'type' argument
-      e.dataTransfer.setData('text', transferData);
-    }
+    e.dataTransfer.setData('text/plain', transferData);
```
Comment (amanmahajan7) [inline src/masks/InteractionMasks.tsx:279]: Same here, an easy fix for IE

**Comment by qili26** [comment]
@amanmahajan7 
I agree with you. Stopping supporting shouldn't make IE 11 critical malfunction (e.g. result in a blank page). I think we shouldn't assume the value is always there. Other than that, this is really a thrilling change!

**Comment by nstepien** [comment]
image
We don't render columns that would be hidden behind frozen columns, so it'd be buggy anyway in IE. Do we really want to support IE11?

**Comment by qili26** [comment]
Just wondering if we should block the frozen feature for IE to not show the incorrect UI? By saying block, I meant when we see the flag is on, we still don't render that feature at all.

**Comment by amanmahajan7** [comment]
We do not want to do extra work to support IE but if we can prevent the page from crashing that would be nice. We would mention in the docs that IE is not supported and IE related bugs will not be addressed. I think it is fine if frozen columns do not work.

**Pull request merged**